File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-26.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        res = l = 0
        count = {}

        for r in range(len(s)):
            if s[r] in count:
                count[s[r]] += 1
            else:
                count[s[r]] = 1
            
            while (r - l + 1) - max(count.values()) > k:
                count[s[l]] -= 1
                l += 1

            res = max(res, r - l + 1)
        
        return res

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-26.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if (nums.length == 0) {
            return 0;
        }
        Arrays.sort(nums);

        int res = 0;
        int curr = nums[0];
        int streak = 0;
        int i = 0;

        while (i < nums.length) {
            // check if we need to reset streak
            if (curr != nums[i]) {
                // reset streak
                curr = nums[i];
                streak = 0;
            }

            // skip over duplicates
            while (i < nums.length && nums[i] == curr) {
                i++;
            }

            streak++;
            curr++;
            
            res = Math.max(res, streak);
        }
        return res;
    }   
}
